File: 백준/Silver/2941. 크로아티아 알파벳/크로아티아 알파벳.cc
#include <iostream>
#include <string>
using namespace std;

int main()
{
    string s;
    cin >> s;
    int count = 0;
    
    for(int i = 0; i < s.length(); i++)
    {
        count++;
        if(s[i] == 'c')
        {
            if(s[i+1] == '=' || s[i+1] == '-')
                count--;
        }
        else if(s[i] == 'j')
        {
            if(s[i-1] == 'l' || s[i-1] == 'n')
                count--;
        }
        else if(s[i] == '=')
        {
            if(s[i-1] == 's' || s[i-1] == 'z')
                count--;
        }
        else if(s[i] == 'd')
        {
            if(s[i+1] == '-')
                count--;
            else if(s[i+1] == 'z')
            {
                if(s[i+2] == '=')
                    count--;
            }
        }     
    }
    cout << count;
    return 0;
    
}

File: 백준/Silver/2941. 크로아티아 알파벳/크로아티아 알파벳.py
lst = ['c=', 'c-', 'dz=', 'd-', 'lj', 'nj', 's=', 'z=']
word = input()
N = len(word)
for i in lst:
    if i in word:
        N = N-(len(i)-1)*word.count(i)
if 'dz=' and 'z=' in word:
    N += word.count('dz=')
print(N)